backend/api/quivr_api/modules/upload/service/upload_file.py

```diff
@@ -23,46 +22,45 @@ def check_file_exists(brain_id: str, file_identifier: str) -> bool:
             file["name"].split(".")[0] == file_identifier.split(".")[0]
             for file in response
         )
-        logger.info(f"File identifier: {file_identifier}")
-        logger.info(f"File exists: {file_exists}")
+        logger.debug(f"File identifier: {file_identifier} exists: {file_exists}")
         if file_exists:
-            logger.info(f"File {file_identifier} exists.")
             return True
         else:
-            logger.info(f"File {file_identifier} does not exist.")
             return False
     except Exception as e:
         logger.error(f"An error occurred while checking the file: {e}")
         return True
 
 
 async def upload_file_storage(
-    supabase_client: AsyncClient,
     file: FileIO | BufferedReader | bytes,
-    file_name: str,
+    storage_path: str,
     upsert: bool = False,
 ):
-    _, file_extension = os.path.splitext(file_name)
-    mime_type, _ = mimetypes.guess_type(file_name)
-    logger.debug(f"Uploading {file_name} to supabase storage.")
+    supabase_client = await get_supabase_async_client()
+    mime_type, _ = mimetypes.guess_type(storage_path)
+    logger.debug(
+        f"Uploading file to {storage_path} using supabase. upsert={upsert}, mimetype={mime_type}"
+    )
 
     if upsert:
-        response = supabase_client.storage.from_("quivr").update(
-            file_name,
+        response = await supabase_client.storage.from_("quivr").update(
+            storage_path,
             file,  # type: ignore
             file_options={
-                "content-type": mime_type or "txt/html",
+                "content-type": mime_type or "application/html",
                 "upsert": "true",
                 "cache-control": "3600",
             },
         )
+        return response
     else:
         try:
             response = await supabase_client.storage.from_("quivr").upload(
-                file_name,
+                storage_path,
                 file,  # type: ignore
                 file_options={
-                    "content-type": mime_type or "text/html",
+                    "content-type": mime_type or "application/html",
```

Comment:
pk cce changement ?

Reply:
This is how mimetypes :  https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types